cime/src/drivers/mct/main/seq_diag_mct.F90

```diff
@@ -2284,57 +2289,29 @@ SUBROUTINE seq_diag_avect_mct(infodata, id, av, dom, gsmap, comment)
           endif
        enddo
```

Comment:
Lift lines between `if` and here to outside of `if ... else ...` block: they're identical?

Reply:
Appears so? Not my code. This is the logic that existed before. I just retained it.

I'll make this change if you want (or you can do it). I don't have an opinion.

Reply:
Thanks Pat. Those were the only questions I had while looking through the changes. I was able to reproduce various scenarios you mentioned in the other issue and the corresponding PR. There's no issue for this PR, but I also think that it's best to catch the NaNs/INFs as early as possible (even if the INF is already fixed by the other PR). I'll make the change. Thanks.

Reply:
Thank you.